receipt_upload/receipt_upload/env_sync/apply.py

```
@@ -0,0 +1,239 @@
+        # AccessDenied / throttling / expired creds etc. are NOT "absent" —
+        # surface them so the caller doesn't silently treat them as missing.
+        raise
+
+
+def execute_migration(
+    plan: MigrationPlan,
+    dst_dynamo,
+    s3,
+    *,
+    apply: bool = False,
+    backup_path: str = None,
```

Ruff (0.15.18)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)